src/brushes/pencil_brush.class.js

```diff
@@ -253,9 +253,11 @@
           newPoints.push(lastPoint);
         }
       }
-      if (newPoints.length === 1) {
-        newPoints.push(new fabric.Point(newPoints[0].x, newPoints[0].y));
-      }
+      /**
```

Comment:
or couldn't we just check in the loop 

```js
if (cDistance >= adjustedDistance || i === l - 1) {
```
?

Reply:
Because with your solution we risk to duplicate the last point in case is already there? or am i reading something wrong?

Reply:
That is also an option. I liked the simplicity of my solution, which actually removed code, but I can see that it may sometimes duplicate the last point. I shall have another think and try some other options.